### PR TITLE
Transcribe mm_ptable_get_attrs_level and rename ptable_lookup

### PR DESCRIPTION
**Pull request by jadephilipoom**
Includes a definition of while loops.

### CODE REVIEW

File: coq-verification/src/Concrete/MM/Implementation.v
Comment (hkim15) [inline coq-verification/src/Concrete/MM/Implementation.v:164]: Just for curiosity. What does N.to_nat() returns if end < begin? And, it is not obvious to me what the first argument (max_iterations := ...) of while_loop does but I can read the details of while_loop in other Coq file later.

**Review by hkim15 — APPROVED** [review]
LGTM.

It is just a nit but (* TODO *) is missing at the end of mm_vm_identity_map's definition.

Definition mm_vm_identity_map
           (s : concrete_state)
           (t : ptable_pointer)
           (begin : paddr_t)
           (end_ : paddr_t)
           (mode : mode_t)
           (ppool : mpool) : (bool * concrete_state * mpool) :=
  (false, s, ppool).

File: coq-verification/src/Concrete/MM/Implementation.v
Comment (hkim15) [reply coq-verification/src/Concrete/MM/Implementation.v]: From now on, I will only approve the PR so that you could rebase & merge by yourself. That would probably make things easier. And, if you don't want to wait for my approval, use "self-approval" at your discretion as well.

**Comment by hkim15** [comment]
From now on, I will only approve the PR so that you could rebase & merge by yourself. That would probably make things easier. And, if you don't want to wait for my approval, use "self-approval" at your discretion as well.

File: coq-verification/src/Concrete/MM/Implementation.v
Comment (jadephilipoom) [reply coq-verification/src/Concrete/MM/Implementation.v:164]: It returns 0 if end < begin, since it's a natural-number type.

the `max_iterations` argument is because Coq does not allow writing definitions that don't obviously or provably terminate. Since it's not obvious to Coq that the incrementing of `begin` with `mm_start_of_next_block` will eventually bring `begin` all the way to `end`, since it doesn't know that for instance `mm_start_of_next_block` doesn't just return `begin` unchanged, I need to somehow guarantee to Coq that the loop will end eventually. The way I'm doing that is passing in a maximum number of iterations, beyond which the loop will break. Since I know that `mm_start_of_next_block` will always increment `begin` by at least 1, it's safe to assume that the loop will reach its termination condition (`begin >= end`) with at most `end - begin` iterations.

**Pull request merged**
